pkg/piecestore/psclient/client.go

```diff
@@ -117,15 +117,37 @@ func (ps *PieceStore) Meta(ctx context.Context, id PieceID) (*pb.PieceSummary, e
 }
 
 // Put uploads a Piece to a piece store Server
-func (ps *PieceStore) Put(ctx context.Context, id PieceID, data io.Reader, ttl time.Time, ba *pb.PayerBandwidthAllocation, authorization *pb.SignedMessage) error {
+func (ps *PieceStore) Put(ctx context.Context, id PieceID, data io.Reader, ttl time.Time, pba *pb.PayerBandwidthAllocation) error {
 	stream, err := ps.client.Store(ctx)
 	if err != nil {
 		return err
 	}
 
+	// Making a copy, otherwise there will be a data race
+	// when another goroutine tries to write the cached size
+	// of this instance at the same time.
+	pbaCopy := &pb.PayerBandwidthAllocation{
```

Comment:
There's probably a shorter way of expressing this... possibly *pbaCopy = *pba?

Reply:
There's a cached atomic variable contained in the structure that is getting written to by other goroutines, so Nat is avoiding reading the whole structure so that that one cached variable doesn't get read, tripping the race detector.

Reply:
I think it definitely needs to be a deep copy, which ends up looking verbose, but I can make it a function to reduce repeated code.